pkg/github/github.go

```diff
@@ -25,11 +25,21 @@ import (
 
 	"github.com/brianvoe/gofakeit/v6"
 	"github.com/google/go-github/v41/github"
+	"github.com/redhat-appstudio/application-service/pkg/metrics"
 	"github.com/redhat-appstudio/application-service/pkg/util"
 )
 
 const AppStudioAppDataOrg = "redhat-appstudio-appdata"
 
+// ServerError is used to identify gitops repo creation failures caused by server errors
+type ServerError struct {
+	err error
+}
+
+func (e *ServerError) Error() string {
+	return fmt.Sprintf("failed to create gitops repo %s", e.err)
```

Comment:
Nit: Can you update the `TestGenerateNewRepository` function to check the error message? Will make codecov happier 🙂